sdks/python/apache_beam/io/kinesis.py

```diff
@@ -0,0 +1,317 @@
+      rate_limit=None,
+      expansion_service=None,
+  ):
+    """
+    Initializes a read operation from Kinesis.
+
+    :param stream_name: Kinesis stream name.
+    :param aws_access_key: Kinesis access key.
+    :param aws_secret_key: Kinesis access key secret.
+    :param region: AWS region. Example: 'us-east-1'.
+    :param service_endpoint: Kinesis service endpoint
+    :param verify_certificate: Enable or disable certificate verification.
+        Never set to False on production. True by default.
+    :param max_num_records: Specifies to read at most a given number of records.
+        Must be greater than 0.
+    :param max_read_time: Specifies to read records during x seconds.
+    :param initial_timestamp_in_stream: Specify reading beginning at the given
+        timestamp in seconds. Must be in the past.
+    :param initial_position_in_stream: Specify reading from some initial
+        position in stream. Possible values:
+        LATEST - Start after the most recent data record (fetch new data).
+        TRIM_HORIZON - Start from the oldest available data record.
+        AT_TIMESTAMP - Start from the record at or after the specified
+        server-side timestamp.
+    :param request_records_limit: Specifies the maximum number of records in
+        GetRecordsResult returned by GetRecords call which is limited by 10K
+        records. If should be adjusted according to average size of data record
+        to prevent shard overloading. More at:
+        docs.aws.amazon.com/kinesis/latest/APIReference/API_GetRecords.html
+    :param up_to_date_threshold: Specifies how late in seconds records consumed
+        by this source can be to still be considered on time. Defaults to zero.
+    :param max_capacity_per_shard: Specifies the maximum number of messages per
+        one shard. Defaults to 10'000.
+    :param watermark_policy: Specifies the watermark policy. Possible values:
+        PROCESSING_TYPE, ARRIVAL_TIME. Defaults to ARRIVAL_TIME.
+    :param watermark_idle_duration_threshold: Use only when watermark policy is
+        ARRIVAL_TIME. Denotes the duration for which the watermark can be idle.
+        Passed in seconds.
+    :param rate_limit: Sets fixed rate policy for given seconds value. By
+        default there is no rate limit.
+    :param expansion_service: The address (host:port) of the ExpansionService.
+    """
+    if watermark_policy:
+      assert watermark_policy == WatermarkPolicy.ARRIVAL_TIME or\
+             watermark_policy == WatermarkPolicy.PROCESSING_TYPE
+
+    if initial_position_in_stream:
+      i = initial_position_in_stream
+      assert i == InitialPositionInStream.AT_TIMESTAMP or \
+             i == InitialPositionInStream.LATEST or \
+             i == InitialPositionInStream.TRIM_HORIZON
+
+    if request_records_limit:
+      assert 0 < request_records_limit <= 10000
+
+    if initial_timestamp_in_stream:
+      assert initial_timestamp_in_stream < time.time()
```

Comment:
This worries me a little bit since the system clock could be wrong. Maybe this should be a warning instead?

Reply:
I think it's a good idea since kinesis java sdk will verify it later anyway. One thing to discuss is whether I should use milliseconds or just seconds to set the Instants in java. For now I've chosen seconds, but maybe millis are always considered as a better option? WDYT?

Reply:
Yeah I think we should prefer millis, since that more closely maps to Instant. It looks like millis is the precision supported by Kinesis based on https://docs.aws.amazon.com/kinesis/latest/APIReference/API_StartingPosition.html

Reply:
Done